cpp/include/cudf/detail/utilities/vector_factories.hpp

```diff
@@ -439,6 +441,40 @@ thrust::host_vector<typename Container::value_type> make_host_vector_sync(
   return make_host_vector_sync(device_span<typename Container::value_type const>{c}, stream);
 }
 
+/**
+ * @brief Asynchronously construct a `host_vector` of the given size
```

Comment:
we should include in the function header that the vector is guaranteed to be backed by pinned memory.